### PR TITLE
adds new property 'manufacturer' for Android (and hardcoded for iOS, Blackberry)

### PR DESCRIPTION
**Pull request by basvanbeek**
For some platforms it makes sense to be able to list the Manufacturer for a device. These changes allow retrieval of manufacturer name for Android, iOS & Blackberry. Have not added to other platforms as I have no access to those or know the API's.

### CODE REVIEW

**Comment by IngusSkaistkalns** [comment]
Real life use-case: App for displaying specific ads on mobile devices in stores. Manager installs this app, thats acts like a screensaver/daydream service, on first app launch, it is registered in administrative portal with all available device info, but without manufacturer. 
So this manager can control what kind of ad-campaign is displaying on any device. Manager creates specific ad-campaign for HTC devices. Wants to assign it to all HTC devices, but no easy way to do that, because there are no HTC devices or Samsung devices, there are at best model names. So to accomplish that, manager have to maintain manufacturer info manually either from administrative portal, or with adding device registration form into application that is displayed on  first app launch. 

Added image for device registration form that gets reviewed by managers after installing app on mobile device and than registering it in administrative portal right from the app, second field would be manufacturer, currently empty and will get replaced with dropdown of available manufacturers or smth like that:
image

**Comment by purplecabbage** [comment]
The user-agent has adequate information to handle this.
-1

**Comment by basvanbeek** [comment]
@purplecabbage so while the Android API has implemented a method to easily get the Manufacturer name from the device, we are declined to reach that method from the most obvious cordova plugin to have this logic disclosed? And the rationale is that in your opinion we just have to parse user-agent strings and build device model to brand mapping tables to get what we want and keep on checking and updating the parsing logic / mapping tables all the time? Highly inefficient and a pure waste of time. 

If the cost of adding, maintaining and having this feature would be high I can understand hesitation but cost is negligible and the cost for developers needing this information and now having to resort to forking the plugin and keeping in sync with ongoing development or even adding an additional plugin to each app requiring this feature sounds very user unfriendly and also discourages people to contribute code if even the simplest of code updates get waved away / neglected like this.

**Comment by jsoref** [comment]
Personally, I don't think exposing this is a good thing. And if the UA already exposes this junk, then, I'd rather people sniff it so that they're aware what they're doing is **wrong**.

-1

**Comment by purplecabbage** [comment]
I am not waiving it away, or neglecting it, just stating my opinion of it's relative value.
If others feel strongly for it then by all means, we should add it.
Generally, I try to resist the urge to add cordova features that are reachable/possible within the browser already.  The overall goal of cordova is to not exist, meaning we shouldn't push developers to depend on cordova specific apis when we don't have to.

All of that said, this is relatively easily supported on wp8 and windows8 so I will have no problem adding it if enough people want it.

File: src/blackberry10/index.js [hunk 1]
```diff
@@ -57,6 +57,7 @@ module.exports = {
             modelName = getModelName(),
             uuid = getUUID(),
             info = {
+                manufacturer: 'Research in Motion',
```
Comment (jsoref) [inline src/blackberry10/index.js]: For reference, `Research in Motion` is not the name of the company anymore, it's `BlackBerry`:

http://press.blackberry.com/press/2013/research-in-motion-changes-its-name-to-blackberry.html

**Comment by IngusSkaistkalns** [comment]
@purplecabbage - maybe, but how do you expect to get string `samsung` from string like this -> `Mozilla/5.0 (Linux; U; Android 4.4.2; en-gb; GT-I9505 Build/KOT49H) AppleWebKit/537.16 (KHTML, like Gecko) Version/4.0 Safari/537.16`, i guess this would require additional maintaining of user-agents or model names in our administrative portal.
Also I don't really understand point of why exposing manufacturer is _wrong_. Am I overlooking some security issues or smth like that?

**Comment by fishgrind** [comment]
Thanks for adding this Bas

**Comment by purplecabbage** [comment]
I have decided to accept this pull request, and implement the windows/wp8 functions.
The fact that the native api makes this information available implies the importance of the info.  Hiding this from cordova apps simply to prevent misuse is probably not helpful. [CB-7285]

**Comment by purplecabbage** [comment]
@basvanbeek Can you sign and submit an ICLA? 
http://cordova.apache.org/#contribute

**Comment by basvanbeek** [comment]
@purplecabbage ICLA signed and sent

**Pull request merged**